livekit-agents/livekit/agents/voice/agent_activity.py

```diff
@@ -1613,10 +1606,7 @@ async def _read_messages(
                         extra={"error": str(e)},
                     )
 
-            if generate_tool_reply and (
-                # no direct cancellation in Gemini
-                GoogleRealtimeModel is None or not isinstance(self.llm, GoogleRealtimeModel)
-            ):
+            if generate_tool_reply and self.llm.capabilities.manual_interruption:
```

Comment:
This isn't for manual_interruption, but more like "should we manually trigger a generation for function tools?)

Reply:
this is indicating the model does not support manual interruption. isn't that not true?

feel free to suggest a different name though